torch/ao/quantization/qconfig_mapping.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import annotations
+from collections import OrderedDict
+from dataclasses import astuple, dataclass
+from typing import Any, Callable, Dict, List, Union
+
+from .qconfig import QConfigAny
+
+
+__all__ = [
+    "QConfigMapping",
+    "QConfigModuleNameEntry",
+    "QConfigModuleNameRegexEntry",
+    "QConfigModuleNameObjectTypeOrderEntry",
+    "QConfigObjectTypeEntry",
+]
+
+
+# TODO: replace all usages with these constants
+GLOBAL_DICT_KEY = ""
+OBJECT_TYPE_DICT_KEY = "object_type"
+MODULE_NAME_REGEX_DICT_KEY = "module_name_regex"
+MODULE_NAME_DICT_KEY = "module_name"
+MODULE_NAME_OBJECT_TYPE_ORDER_DICT_KEY = "module_name_object_type_order"
+
+
+@dataclass
+class QConfigObjectTypeEntry:
+    # object_type can be
+    # 1. module type (call_module)
+    # 2. function (call_function)
+    # 3. string (call_method)
+    object_type: Union[Callable, str]
+    qconfig: QConfigAny
+
+
+@dataclass
+class QConfigModuleNameRegexEntry:
+    module_name_regex: str
+    qconfig: QConfigAny
+
+
+@dataclass
+class QConfigModuleNameEntry:
+    module_name: str
+    qconfig: QConfigAny
+
+
+@dataclass
+class QConfigModuleNameObjectTypeOrderEntry:
+    module_name: str
+    object_type: Callable
+    index: int
+    qconfig: QConfigAny
+
+
+class QConfigMapping:
```

Comment:
it would be nice to have rendered documentation for this class, can be in a separate PR